src/CredentialsLoader.php

```diff
@@ -169,4 +169,13 @@ public function updateMetadata(
     $metadata_copy[self::AUTH_METADATA_KEY] = array('Bearer ' . $result['access_token']);
     return $metadata_copy;
   }
+
+  /**
+   * Gets the underlying `OAuth2` instance used to conduct authorization.
+   *
+   * @return OAuth2
+   */
+  public function getAuth() {
```

Comment:
I'm not a huge fan of exposing this unless it's 100% necessary. I think the main issue here is there's no way to get at your access token without using `Cache` classes, yes?